modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,375 @@
+<?php
+/**
+ * Module Name: Dominant Color
+ * Description: Adds support to store dominant-color for an image and create a placeholder background with that color.
+ * Experimental: Yes
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @param array $metadata The attachment metadata.
+ * @param int   $attachment_id The attachment ID.
+ *
+ * @return array $metadata
+ * @since n.e.x.t
+ */
+function dominant_color_metadata( $metadata, $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+
+		return $metadata;
+	}
+
+	$dominant_color = dominant_color_get( $attachment_id );
+
+	if ( ! empty( $dominant_color ) ) {
+		$metadata['dominant_color'] = $dominant_color;
+	}
+
+	return $metadata;
+}
+
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_metadata', 10, 2 );
+
+
+/**
+ * Add the dominant color metadata to the attachment.
+ *
+ * @param array $metadata Metadata for the attachment.
+ * @param int   $attachment_id attachement id.
+ *
+ * @return array $metadata
+ * @since n.e.x.t
+ */
+function dominant_color_has_transparency_metadata( $metadata, $attachment_id ) {
+
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+		return $metadata;
+	}
+
+	$has_transparency = dominant_color_get_has_transparency( $attachment_id );
+
+	if ( ! empty( $has_transparency ) ) {
+		$metadata['has_transparency'] = $has_transparency;
+	} else {
+		$metadata['has_transparency'] = false;
+	}
+
+	return $metadata;
+}
+
+add_filter( 'wp_generate_attachment_metadata', 'dominant_color_has_transparency_metadata', 10, 2 );
+
+/**
+ * Filter various image attributes to add the dominant color to the image
+ *
+ * @param array  $attr Attributes for the image markup.
+ * @param object $attachment Image attachment post.
+ *
+ * @return mixed
+ * @since n.e.x.t
+ */
+function dominant_color_tag_add_adjust_to_image_attributes( $attr, $attachment ) {
```

Comment:
See my other comments about defaults: I think this function would be the right place to apply any defaults. Basically, if `dominant_color` or `has_transparency` is not set, set the reasonable default _here_. This way the metadata itself remains accurate and free of unnecessarily stored defaults, and defaults are only used on the fly as a true fallback.